lib/Github/Client.php

```diff
@@ -274,15 +352,42 @@ public function setHttpClient(HttpClientInterface $httpClient)
      */
     public function clearHeaders()
     {
-        $this->getHttpClient()->clearHeaders();
+        $this->headers = array(
+            'Accept' => sprintf('application/vnd.github.%s+json', $this->options['api_version']),
+            'User-Agent' => sprintf('%s', $this->options['user_agent']),
+        );
+
+        $headersPlugin = new Plugin\HeaderAppendPlugin($this->headers);
+        $this->removePlugin(Plugin\HeaderAppendPlugin::class);
+        $this->addPlugin($headersPlugin);
     }
 
     /**
      * @param array $headers
      */
-    public function setHeaders(array $headers)
+    public function addHeaders(array $headers)
```

Comment:
I was confused by the name. `set*` is usually replacing the existing value. Since we are not replacing the value I change it to `add*`. 

But this is an opinionated topic.